src/watertap_contrib/reflo/unit_models/zero_order/forward_osmosis_zo.py

```diff
@@ -0,0 +1,886 @@
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+
+        opt = get_solver(solver, optarg)
+        # ---------------------------------------------------------------------
+        flags = blk.feed_props.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args,
+            hold_state=True,
+        )
+        init_log.info("Initialization Step 1 Complete.")
+        # ---------------------------------------------------------------------
+        # Initialize other state blocks
+        if state_args is None:
+            blk.state_args = state_args = {}
+            state_dict = blk.feed_props[
+                blk.flowsheet().config.time.first()
+            ].define_port_members()
+
+            for k in state_dict.keys():
+                if state_dict[k].is_indexed():
+                    state_args[k] = {}
+                    for m in state_dict[k].keys():
+                        state_args[k][m] = state_dict[k][m].value
+                else:
+                    state_args[k] = state_dict[k].value
+
+        # Initial guess of brine properties
+        state_args_brine = deepcopy(state_args)
+        for p, j in blk.brine_props.phase_component_set:
+            if p == "Liq" and j == "H2O":
+                state_args_brine["flow_mass_phase_comp"][(p, j)] = (
+                    state_args["flow_mass_phase_comp"][(p, j)]
+                    * (1 - blk.recovery_ratio / blk.nanofiltration_recovery_ratio)
+                    * pyunits.kg
+                    / pyunits.s
+                )
+
+        blk.brine_props.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args_brine,
+        )
+
+        # Initialize draw solution blocks
+        state_args_draw_solution = {}
+        state_dict_draw_solution = blk.strong_draw_props[
+            blk.flowsheet().config.time.first()
+        ].define_port_members()
+
+        for k in state_dict_draw_solution.keys():
+            if state_dict_draw_solution[k].is_indexed():
+                state_args_draw_solution[k] = {}
+                for m in state_dict_draw_solution[k].keys():
+                    state_args_draw_solution[k][m] = state_dict_draw_solution[k][
+                        m
+                    ].value
+            else:
+                state_args_draw_solution[k] = state_dict_draw_solution[k].value
+
+        # Initial guess of weak draw properties
+        state_args_weak_draw = deepcopy(state_args_draw_solution)
+        for p, j in blk.weak_draw_props.phase_component_set:
+            if p == "Liq" and j == "H2O":
+                state_args_weak_draw["flow_mass_phase_comp"][(p, j)] = (
+                    state_args["flow_mass_phase_comp"][(p, j)]
+                    * blk.recovery_ratio
+                    / blk.nanofiltration_recovery_ratio
+                    * 1.5
```

Comment:
What is this number? The initialization routine looks good and I liked that you called out what some of these other constants in the routine are for.

Reply:
The comment is added now.